test/src/test/java/hudson/model/ItemGroupMixInTest.java

```diff
@@ -47,4 +65,139 @@
         assertEquals("valid", items.iterator().next().getName());
     }
 
+  /**
+   * This test unit makes sure that if part of the config.xml file is
+   * deleted it will still load everything else inside the folder.
+   * The test unit expects an IOException is thrown, and the one failed
+   * job fails to load.
+   */
+  @Issue("JENKINS-22811")
+  @Test
+  public void xmlFileFailsToLoad() throws Exception {
+    MockFolder folder = r.createFolder("folder");
+    assertNotNull(folder);
+
+    AbstractProject project = folder.createProject(FreeStyleProject.class, "job1");
+    AbstractProject project2 = folder.createProject(FreeStyleProject.class, "job2");
+    AbstractProject project3 = folder.createProject(FreeStyleProject.class, "job3");
+
+    File configFile = project.getConfigFile().getFile();
+
+    List<String> lines = FileUtils.readLines(configFile).subList(0, 5);
+    configFile.delete();
+
+    // Remove half of the config.xml file to make "invalid" or fail to load
+    FileUtils.writeByteArrayToFile(configFile, lines.toString().getBytes());
+    for (int i = lines.size() / 2; i < lines.size(); i++) {
+      FileUtils.writeStringToFile(configFile, lines.get(i), true);
+    }
+
+    // Reload Jenkins.
+    r.jenkins.reload();
+
+    // Folder
+    assertNotNull("Folder failed to load.", r.jenkins.getItemByFullName("folder"));
+    assertNull("Job should have failed to load.", r.jenkins.getItemByFullName("folder/job1"));
+    assertNotNull("Other job in folder should have loaded.", r.jenkins.getItemByFullName("folder/job2"));
+    assertNotNull("Other job in folder should have loaded.", r.jenkins.getItemByFullName("folder/job3"));
+  }
+
+  /**
+   * This test unit makes sure that any exception that is thrown is caught, however the folder
+   * will continue to load not impacting any other jobs in the folder.
+   */
+  @LocalData
+  @Issue("JENKINS-22811")
+  @Test
+  public void xmlFileReadExceptionOnLoad() throws Exception {
+    MockFolder d = r.jenkins.getItemByFullName("d", MockFolder.class);
+    assertNotNull(d);
+    Collection<TopLevelItem> items = d.getItems();
+    assertEquals(5, items.size());
+    Iterator<TopLevelItem> iterator = items.iterator();
+    assertEquals("badBuildStep", iterator.next().getName());
+    assertEquals("badBuildTrigger", iterator.next().getName());
+    assertEquals("badBuildWrapper", iterator.next().getName());
+    assertEquals("badPublisher", iterator.next().getName());
+    assertEquals("valid", iterator.next().getName());
```

Comment:
There is no guaranteed order for `ItemGroup.getItems`, so this test idiom may not be used. I would recommend something like

```
Set<String> names = new TreeSet<String>();
for (Item i : d.getItems()) {
    names.add(i.getName());
}
assertEquals("[badBuildStep, badBuildTrigger, badBuildWrapper, badPublisher, valid]", names.toString());
```

although simply asserting the size of the folder would probably be enough to make sure all the jobs got loaded.